lib/modules/deployment/contract_deployer.js

```diff
@@ -213,7 +212,9 @@ class ContractDeployer {
         });
       },
       function applyBeforeDeploy(next) {
-        self.plugins.emitAndRunActionsForEvent('deploy:contract:beforeDeploy', {contract: contract}, next);
+        self.plugins.emitAndRunActionsForEvent('deploy:contract:beforeDeploy', {contract: contract}, () => {
+          next();
```

Comment:
Why?

Reply:
the callback returns params, which mess up the waterfall

Reply:
Could you put `(_params) => {` just so we know